src/scales/scale.time.js

```diff
@@ -281,6 +281,29 @@ module.exports = function(Chart) {
 				max = ticks.length ? ticks[ticks.length - 1] : max;
 			}
 
+			// If the chart has bars, shift min and max based on the sample size
```

Comment:
I think you'll need a different logic here if `mode: series`

`ticks.mode` (`'linear'`(default)|`series`): `series` displays ticks at the same distance from each other, whatever the time value they represent, while `linear` displays them linearly in time: the distance between each tick represent the amount of time between their time values.